src/main/java/gregtech/loaders/dungeon/ChestGenHooks.java

```diff
@@ -84,8 +81,70 @@ public ItemStack apply(@Nonnull ItemStack stack, @Nonnull Random rand, @Nonnull
     }
 
     public static void addRolls(ResourceLocation tableLocation, int minAdd, int maxAdd) {
-        rollVals.put(tableLocation, new RandomValueRange(minAdd, maxAdd));
+        rollValues.put(tableLocation, new RandomValueRange(minAdd, maxAdd));
+    }
+
+    private static final ItemStackHashStrategy HASH_STRATEGY = ItemStackHashStrategy.comparingAllButCount();
+
+    private static @NotNull String createEntryName(@NotNull ItemStack stack, @NotNull String modid, int weight,
+                                                   @NotNull RandomWeightLootFunction function) {
+        int hashCode = Objects.hash(HASH_STRATEGY.hashCode(stack), modid, weight, function.getMinAmount(), function.getMaxAmount());
+        return String.format("#%s:loot_%s", modid, hashCode);
+    }
+
+    private static class GTLootEntryItem extends LootEntryItem {
+
+        private final ItemStack stack;
+
+        public GTLootEntryItem(@NotNull ItemStack stack, int weight, LootFunction lootFunction, @NotNull String entryName) {
+            super(stack.getItem(), weight, 1, new LootFunction[]{lootFunction}, NO_CONDITIONS, entryName);
+            this.stack = stack;
+        }
+
+        @Override
+        public @NotNull String toString() {
+            return "GTLootEntryItem{name=" + getEntryName() + ", stack=" + GTStringUtils.itemStackToString(stack) + '}';
+        }
     }
 
+    private static class RandomWeightLootFunction extends LootFunction {
+
+        private final ItemStack stack;
+        private final int minAmount;
+        private final int maxAmount;
+
+        public RandomWeightLootFunction(@NotNull ItemStack stack, int minAmount, int maxAmount) {
+            super(NO_CONDITIONS);
+            Preconditions.checkArgument(minAmount <= maxAmount, "minAmount must be <= maxAmount");
+            this.stack = stack;
+            this.minAmount = minAmount;
+            this.maxAmount = maxAmount;
+        }
+
+        public int getMinAmount() {
+            return minAmount;
+        }
 
+        public int getMaxAmount() {
+            return maxAmount;
+        }
+
+        @Override
+        public @NotNull ItemStack apply(@NotNull ItemStack itemStack, @NotNull Random rand, @NotNull LootContext context) {
+            itemStack.setItemDamage(stack.getItemDamage());
+            NBTTagCompound tagCompound = stack.getTagCompound();
+            if (tagCompound != null) {
+                itemStack.setTagCompound(tagCompound.copy());
+            }
+
+            if (minAmount == maxAmount) {
+                itemStack.setCount(1);
```

Comment:
This should set to the minAmount. It is not guaranteed that if min = max that both are equal to 1